client/lib/signup/step-actions/fetch-sites-and-user.js

```diff
@@ -14,7 +12,8 @@ async function fetchSitesUntilSiteAppears( siteSlug, reduxStore ) {
 }
 
 export function fetchSitesAndUser( siteSlug, onComplete, reduxStore ) {
-	Promise.all( [ fetchSitesUntilSiteAppears( siteSlug, reduxStore ), user().fetch() ] ).then(
-		onComplete
-	);
+	Promise.all( [
+		fetchSitesUntilSiteAppears( siteSlug, reduxStore ),
+		reduxStore.dispatch( fetchCurrentUser() ),
```

Comment:
The result of some discussion in Slack about signup hanging: `user().fetch()` returns a Promise but AFAIK this does not, so won't this not work?

Reply:
We did improve this in #53814 but we might need to also return the `fetchingUser` fetching promise at the end of the thunk. Won't that fix the issue?

Reply:
I'll give that a try on Monday. Feel free to merge the revert.